pkg/server/status.go

```diff
@@ -1539,6 +1539,25 @@ func (s *statusServer) Stacks(
 	return stacksLocal(req)
 }
 
+// Nodes returns all node statuses for a secondary tenant with
+// `can_view_node_info` capability.
```

Comment:
Do we have a test that covers that this endpoint fails without the capability? If not, it may be worth it to add one.

Reply:
I was using the test mentioned in the ticket that was failing before my changes. I tried fixing the test to work for negative case too but found out `can_view_node_info` has no affect on shared tenants. I tested the changes for "external" secondary tenants and it was works as expected. 

On digging further, I found https://github.com/cockroachdb/cockroach/blob/75a1f36daeebcaa1bf9333f1912b83c2b7a54472/pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go#L469-L475 

From the above comment it looks like shared tenants implicitly have all capabilities unless we introduce a new mode which I believe will be a bigger change.

Reply:
Yup, sorry for not being a bit more explicit. I think we can probably write a test that explicitly starts a separate process tenant and then makes this API call.  You can do this either by passing `ExternalTestTenantAlwaysEnabled` as the value of `DefaultTestTenant` during TestServer startup, or by passing `TestControlsTenantsExplicitly` and then using the methods on TestServer to start a separate process test server manually.  Putting this test in place in the server package will probably be useful as we can add more of these negative cases to it as we enable more endpoints.

Reply:
https://github.com/cockroachdb/cockroach/blob/b617593ae2d5013035b3633f95c585948471d2d3/pkg/server/storage_api/nodes_test.go#L93-L103
I created the negative test for external tenant using a table. Let me know if you prefer a separate test for the external mode.